dfastmi/io/RiverParameterData.py

```diff
@@ -163,54 +184,40 @@ def collect_values_logical(
             branch one bool or a list of booleans depending on input argument nval.
         """
         self.is_initialized()
-        try:
-            g_val = self._config["General"][key]
-        except:
-            g_val = ""
-        all_values = []
-        for ib in range(len(self._branches)):
-            branch = self._branches[ib]
+        val = self.__read_value(key, branch, reach)
+        if val == "" and default is not None:
+            bval = default
+        else:
             try:
-                b_val = self._config[branch][key]
+                vals = tuple(x.lower() in ['true', '1', 't', 'y', 'yes'] for x in val.split())
```

Comment:
maybe good to move this try catch to a seperate method with a descriptive name of what it does and alway return  vals[0]?

Reply:
I hae the feeling this piece of code can be improved, but I also have the feeling that the piece of code doesnt function as it should.
Since vals is a tuple, but bval which is returned is always only the first value of the tuple, making all other parts of the tuple useless...

Reply:
Discussed, we will leave these comments for a later refactoring.
This will be tackled in https://issuetracker.deltares.nl/browse/DFAST-112